src/Core/Managed/Shared/Extensibility/Implementation/Clock.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Diagnostics;
+
+    /// <summary>
+    /// A highly-accurate, precise and testable clock.
+    /// </summary>
+    internal class Clock : IClock
+    {
+        // DateTimeOffset.Now is accurate to ~16ms. We want accurate and precise timestamps,
+        // so use Stopwatch which is based on perf counters and is highly accurate and precise.
+        // Making a static instance of Stopwatch shared by all operations.        
+        private static readonly Stopwatch InitialStopWatch = Stopwatch.StartNew();
```

Comment:
I don't think it needs a fancy name, it's private so it's not "shared" in the context of the class.